backend/open_api_spec.yml

```diff
@@ -0,0 +1,829 @@
+                        type: array
+                        items:
+                          $ref: "#/components/schemas/collection_with_dataset_preview"
+        "401":
+          $ref: "#/components/responses/401"
+    post:
+      description: Create a new Collection
+      security:
+        - cxguserCookie: []
+      requestBody:
+        content:
+          application/json:
+            $ref: "#/components/schemas/collection_metadata"
+      responses:
+        "201":
+          description: Created
+            content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/collection_uuid"
+        "401":
+          $ref: "#/components/responses/401"
+
+  /curation/collections/{collection_id}:
+    delete:
+      description: Delete a private Collection. Delete an existing revision for a public Collection.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              type: string
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    get:
+      description: Fetch Collection metadata and associated Datasets, public and private.
+      security:
+        - cxgUserCookieLenient: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/query_visibility"
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              $ref: "#/components/schemas/collection"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+    put:
+      description: Start a revision. Update any included Collection metadata fields.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+      requestBody:
+        content:
+          application/json:
+            $ref: "#/components/schemas/collection_metadata"
+      responses:
+        "204":
+          $ref: "#/components/responses/204"
+        "401":
+          $ref: "#/components/responses/401"
+        "404":
+          $ref: "#/components/responses/404"
+
+  /curation/collections/{collection_id}/datasets:
+    delete:
+      description: >
+        Delete a private Dataset. For a public Dataset, start a revision if
+        one does not exist, then delete the Dataset from the revision.
+      security:
+        - cxguserCookie: []
+      parameters:
+        - $ref: "#/components/parameters/path_collection_uuid"
+        - $ref: "#/components/parameters/optional_query_curator_tag"
+        - $ref: "#/components/parameters/optional_query_dataset_uuid"
+        - $ref: "#/components/parameters/query_visibility"
```

Comment:
Must be able to specify visibility so that they can both
1) delete a dataset from an ongoing revision `?visibility=private`
and
2) start a revision, and delete the dataset from the revision `?visibility=public`

Reply:
Alternatively we can outlaw #2 and force them to create a revision via PUT to `/curation/collections/{collection_id}/`

Reply:
forbid deletion of dataset from public collection